rasa/nlu/components.py

```diff
@@ -629,17 +635,25 @@ def can_handle_language(cls, language: Hashable) -> bool:
             `True` if component can handle specific language, `False` otherwise.
         """
 
-        # if language_list is set to `None` it means: support all languages
+        # If both `supported_language_list` and `not_supported_language_list` are set to `None`,
+        # it means: support all languages
         if language is None or (
             cls.supported_language_list is None
             and cls.not_supported_language_list is None
         ):
             return True
 
-        language_list = cls.supported_language_list or []
-        not_supported_language_list = cls.not_supported_language_list or []
+        # check language supporting settings
+        if cls.supported_language_list and cls.not_supported_language_list:
+            # When user set both language supporting settings to not None, it will lead to ambiguity.
+            raise Exception(
+                "Only one of `supported_language_list` and `not_supported_language_list` can be set to not None"
+            )
 
-        return language in language_list or language not in not_supported_language_list
+        if cls.supported_language_list:
+            return language in cls.supported_language_list
+        else:
+            return language not in cls.not_supported_language_list
```

Comment:
This check will result in a runtime exception if `not_supported_language_list` was set to `[]` and `supported_language_list` was `None`. I know that is not really something someone would set, but we can catch the case by either - 
1. changing `else` construct with `elif cls.not_supported_language_list` or
2. Turning both `cls.supported_language_list` and `cls.not_supported_language_list` to empty lists if they are none and then proceeding with the above `if` constructs.

Reply:
I add some guard clause for this in a later commit.